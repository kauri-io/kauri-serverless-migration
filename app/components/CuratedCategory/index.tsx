import React from 'react'
import styled from 'styled-components'
import { Label, BodyCard } from '../Typography'
import Image from '../Image'
import theme from '../../lib/theme-config'

const DEFAULT_CARD_WIDTH = theme.DEFAULT_CARD_WIDTH

const Container = styled.div<{ background: null | string }>`
    display: flex;
    z-index: 1;
    position: relative;
    flex-direction: column;
    width: ${DEFAULT_CARD_WIDTH}px;
    height: 90px;
    background: ${props => props.theme.colors.bgPrimary};
    cursor: pointer;
    border-radius: 4px;
    @media (max-width: ${props => props.theme.breakpoints[0]}) {
        height: 60px;
        width: 170px;
        > a {
            z-index: 9001;
        }
        > a > div > *:last-child {
            display: none;
        }
    }
`

const Content = styled.div`
    display: flex;
    z-index: 9001;
    width: 100%;
    position: absolute;
    flex-direction: column;
    text-overflow: ellipsis;
    padding: ${props => props.theme.space[2]}px;
    overflow-x: hidden;
    > :first-child {
        margin-bottom: ${props => props.theme.space[1]}px;
    }
`

export const CuratedCategoriesSection = styled.section`
    display: grid;
    grid-template-columns: repeat(
        auto-fill,
        minmax(${DEFAULT_CARD_WIDTH}px, ${DEFAULT_CARD_WIDTH}px)
    );
    width: 100%;
    grid-gap: ${props => props.theme.space[1]}px
        ${props => props.theme.space[3]}px;
    padding-top: ${props => props.theme.space[3]}px;
    padding-bottom: 0px;
    ${props => props.theme.padContent};
    @media (max-width: ${props => props.theme.breakpoints[0]}) {
        grid-template-columns: repeat(auto-fill, minmax(170px, 170px));
        justify-content: center;
    }
`

interface IProps {
    category: string
    description: string
    background: string | null
    link?: string
}

const CuratedCategory: React.FunctionComponent<IProps> = ({
    category,
    description,
    background,
    link,
}) => (
    <Container background={background}>
        {background && (
            <Image
                asBackground={true}
                height="90px"
                width={`${DEFAULT_CARD_WIDTH}px`}
                mobileHeight="60px"
                mobileWidth="170px"
                overlay={{ color: theme.bgPrimary, opacity: 0.7 }}
                image={background}
                borderRadius={'4px'}
            />
        )}
        <a href={link}>
            <Content>
                <Label textTransform="uppercase" color="white">
                    {category}
                </Label>
                <BodyCard color="white">{description}</BodyCard>
            </Content>
        </a>
    </Container>
)

export default CuratedCategory
