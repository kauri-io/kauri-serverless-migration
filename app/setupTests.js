import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import theme from './lib/theme-config'
import { ThemeProvider } from 'styled-components'

Enzyme.configure({ adapter: new Adapter() })

const observeMock = {
    observe: () => null,
    disconnect: () => null, // maybe not needed
}

window.IntersectionObserver = () => observeMock

export const mountWithTheme = children =>
    mount(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const renderWithTheme = children =>
    render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const shallowWithTheme = children =>
    shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
