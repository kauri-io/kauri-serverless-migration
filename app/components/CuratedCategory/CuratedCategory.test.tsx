import CuratedCategory from './index'
import { mountWithTheme } from '../../setupTests'

let props = {
    linkComponent: (e: any) => e,
    category: 'test',
    description: 'test',
    background: 'https://test.com/image.png',
}

describe('components/CuratedCategory', () => {
    it('snapshot should match', () => {
        const wrapper = mountWithTheme(<CuratedCategory {...props} />)
        expect(wrapper).toMatchSnapshot()
    })
})
