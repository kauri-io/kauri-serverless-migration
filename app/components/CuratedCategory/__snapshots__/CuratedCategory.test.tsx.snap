// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CuratedCategory should match snapshot 1`] = `
<ThemeProvider
  theme={
    Object {
      "breakpoints": Object {
        "between": [Function],
        "down": [Function],
        "keys": Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ],
        "only": [Function],
        "up": [Function],
        "values": Object {
          "lg": 1280,
          "md": 960,
          "sm": 600,
          "xl": 1920,
          "xs": 0,
        },
        "width": [Function],
      },
      "direction": "ltr",
      "mixins": Object {
        "gutters": [Function],
        "toolbar": Object {
          "@media (min-width:0px) and (orientation: landscape)": Object {
            "minHeight": 48,
          },
          "@media (min-width:600px)": Object {
            "minHeight": 64,
          },
          "minHeight": 56,
        },
      },
      "overrides": Object {},
      "palette": Object {
        "action": Object {
          "active": "rgba(0, 0, 0, 0.54)",
          "disabled": "rgba(0, 0, 0, 0.26)",
          "disabledBackground": "rgba(0, 0, 0, 0.12)",
          "hover": "rgba(0, 0, 0, 0.08)",
          "hoverOpacity": 0.08,
          "selected": "rgba(0, 0, 0, 0.14)",
        },
        "augmentColor": [Function],
        "background": Object {
          "default": "#f5f5f5",
          "paper": "#fff",
        },
        "common": Object {
          "black": "#1E2428",
          "white": "#fff",
        },
        "contrastThreshold": 3,
        "divider": "rgba(0, 0, 0, 0.12)",
        "error": Object {
          "contrastText": "#fff",
          "dark": "#d32f2f",
          "light": "#e57373",
          "main": "#f44336",
        },
        "getContrastText": [Function],
        "grey": Object {
          "100": "#f5f5f5",
          "200": "#eeeeee",
          "300": "#e0e0e0",
          "400": "#bdbdbd",
          "50": "#fafafa",
          "500": "#9e9e9e",
          "600": "#757575",
          "700": "#616161",
          "800": "#424242",
          "900": "#212121",
          "A100": "#d5d5d5",
          "A200": "#aaaaaa",
          "A400": "#303030",
          "A700": "#616161",
        },
        "primary": Object {
          "contrastText": "rgba(0, 0, 0, 0.87)",
          "dark": "rgb(7, 118, 93)",
          "light": "rgb(59, 186, 158)",
          "main": "#0BA986",
        },
        "secondary": Object {
          "contrastText": "rgba(0, 0, 0, 0.87)",
          "dark": "rgb(178, 178, 178)",
          "light": "rgb(255, 255, 255)",
          "main": "#ffffff",
        },
        "text": Object {
          "disabled": "rgba(0, 0, 0, 0.38)",
          "hint": "rgba(0, 0, 0, 0.38)",
          "primary": "rgba(0, 0, 0, 0.87)",
          "secondary": "rgba(0, 0, 0, 0.54)",
        },
        "tonalOffset": 0.2,
        "type": "light",
      },
      "props": Object {},
      "shadows": Array [
        "none",
        "0px 1px 3px 0px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 2px 1px -1px rgba(0,0,0,0.12)",
        "0px 1px 5px 0px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 3px 1px -2px rgba(0,0,0,0.12)",
        "0px 1px 8px 0px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 3px 3px -2px rgba(0,0,0,0.12)",
        "0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12)",
        "0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12)",
        "0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12)",
        "0px 4px 5px -2px rgba(0,0,0,0.2),0px 7px 10px 1px rgba(0,0,0,0.14),0px 2px 16px 1px rgba(0,0,0,0.12)",
        "0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12)",
        "0px 5px 6px -3px rgba(0,0,0,0.2),0px 9px 12px 1px rgba(0,0,0,0.14),0px 3px 16px 2px rgba(0,0,0,0.12)",
        "0px 6px 6px -3px rgba(0,0,0,0.2),0px 10px 14px 1px rgba(0,0,0,0.14),0px 4px 18px 3px rgba(0,0,0,0.12)",
        "0px 6px 7px -4px rgba(0,0,0,0.2),0px 11px 15px 1px rgba(0,0,0,0.14),0px 4px 20px 3px rgba(0,0,0,0.12)",
        "0px 7px 8px -4px rgba(0,0,0,0.2),0px 12px 17px 2px rgba(0,0,0,0.14),0px 5px 22px 4px rgba(0,0,0,0.12)",
        "0px 7px 8px -4px rgba(0,0,0,0.2),0px 13px 19px 2px rgba(0,0,0,0.14),0px 5px 24px 4px rgba(0,0,0,0.12)",
        "0px 7px 9px -4px rgba(0,0,0,0.2),0px 14px 21px 2px rgba(0,0,0,0.14),0px 5px 26px 4px rgba(0,0,0,0.12)",
        "0px 8px 9px -5px rgba(0,0,0,0.2),0px 15px 22px 2px rgba(0,0,0,0.14),0px 6px 28px 5px rgba(0,0,0,0.12)",
        "0px 8px 10px -5px rgba(0,0,0,0.2),0px 16px 24px 2px rgba(0,0,0,0.14),0px 6px 30px 5px rgba(0,0,0,0.12)",
        "0px 8px 11px -5px rgba(0,0,0,0.2),0px 17px 26px 2px rgba(0,0,0,0.14),0px 6px 32px 5px rgba(0,0,0,0.12)",
        "0px 9px 11px -5px rgba(0,0,0,0.2),0px 18px 28px 2px rgba(0,0,0,0.14),0px 7px 34px 6px rgba(0,0,0,0.12)",
        "0px 9px 12px -6px rgba(0,0,0,0.2),0px 19px 29px 2px rgba(0,0,0,0.14),0px 7px 36px 6px rgba(0,0,0,0.12)",
        "0px 10px 13px -6px rgba(0,0,0,0.2),0px 20px 31px 3px rgba(0,0,0,0.14),0px 8px 38px 7px rgba(0,0,0,0.12)",
        "0px 10px 13px -6px rgba(0,0,0,0.2),0px 21px 33px 3px rgba(0,0,0,0.14),0px 8px 40px 7px rgba(0,0,0,0.12)",
        "0px 10px 14px -6px rgba(0,0,0,0.2),0px 22px 35px 3px rgba(0,0,0,0.14),0px 8px 42px 7px rgba(0,0,0,0.12)",
        "0px 11px 14px -7px rgba(0,0,0,0.2),0px 23px 36px 3px rgba(0,0,0,0.14),0px 9px 44px 8px rgba(0,0,0,0.12)",
        "0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12)",
      ],
      "shape": Object {
        "borderRadius": 4,
      },
      "spacing": [Function],
      "transitions": Object {
        "create": [Function],
        "duration": Object {
          "complex": 375,
          "enteringScreen": 225,
          "leavingScreen": 195,
          "short": 250,
          "shorter": 200,
          "shortest": 150,
          "standard": 300,
        },
        "easing": Object {
          "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
          "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
          "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
          "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
        },
        "getAutoHeightDuration": [Function],
      },
      "typography": Object {
        "body1": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
          "fontSize": "1rem",
          "fontWeight": 400,
          "letterSpacing": "0.00938em",
          "lineHeight": 1.5,
        },
        "body2": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
          "fontSize": "0.875rem",
          "fontWeight": 400,
          "letterSpacing": "0.01071em",
          "lineHeight": 1.43,
        },
        "button": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
          "fontSize": "0.875rem",
          "fontWeight": 500,
          "letterSpacing": "0.02857em",
          "lineHeight": 1.75,
          "textTransform": "uppercase",
        },
        "caption": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
          "fontSize": "0.75rem",
          "fontWeight": 400,
          "letterSpacing": "0.03333em",
          "lineHeight": 1.66,
        },
        "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
        "fontSize": 14,
        "fontWeightBold": 700,
        "fontWeightLight": 300,
        "fontWeightMedium": 500,
        "fontWeightRegular": 400,
        "h1": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
          "fontSize": "6rem",
          "fontWeight": 300,
          "letterSpacing": "-0.01562em",
          "lineHeight": 1,
        },
        "h2": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
          "fontSize": "3.75rem",
          "fontWeight": 300,
          "letterSpacing": "-0.00833em",
          "lineHeight": 1,
        },
        "h3": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
          "fontSize": "3rem",
          "fontWeight": 400,
          "letterSpacing": "0em",
          "lineHeight": 1.04,
        },
        "h4": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
          "fontSize": "2.125rem",
          "fontWeight": 400,
          "letterSpacing": "0.00735em",
          "lineHeight": 1.17,
        },
        "h5": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
          "fontSize": "1.5rem",
          "fontWeight": 400,
          "letterSpacing": "0em",
          "lineHeight": 1.33,
        },
        "h6": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
          "fontSize": "1.25rem",
          "fontWeight": 500,
          "letterSpacing": "0.0075em",
          "lineHeight": 1.6,
        },
        "htmlFontSize": 16,
        "overline": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
          "fontSize": "0.75rem",
          "fontWeight": 400,
          "letterSpacing": "0.08333em",
          "lineHeight": 2.66,
          "textTransform": "uppercase",
        },
        "pxToRem": [Function],
        "round": [Function],
        "subtitle1": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
          "fontSize": "1rem",
          "fontWeight": 400,
          "letterSpacing": "0.00938em",
          "lineHeight": 1.75,
        },
        "subtitle2": Object {
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
          "fontSize": "0.875rem",
          "fontWeight": 500,
          "letterSpacing": "0.00714em",
          "lineHeight": 1.57,
        },
      },
      "zIndex": Object {
        "appBar": 1100,
        "drawer": 1200,
        "mobileStepper": 1000,
        "modal": 1300,
        "snackbar": 1400,
        "tooltip": 1500,
      },
    }
  }
>
  <ThemeProvider
    theme={
      Object {
        "DEFAULT_CARD_WIDTH": 288,
        "MAIN_BODY_WIDTH": 1242,
        "bg": Object {
          "bgPrimary": "#1E2428",
          "bgSecondary": "#1E3D3B",
          "borderTextArea": "#979797",
          "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
          "disabledBackgroundColor": "#D6D6D6",
          "disabledTextColor": "#9B9B9B",
          "divider": "#EBEBEB",
          "errorRedColor": "#C03030",
          "hoverTextColor": "#108E72",
          "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
          "paddingTop": "2.5em",
          "primary": "#0BA986",
          "primaryColor": "#0BA986",
          "primaryDark": "#11856B",
          "primaryTextColor": "#1E2428",
          "secondary": "#11856B",
          "secondaryBlue": "rgb(81, 123, 162, 1)",
          "secondaryBlueDark": "rgb(81, 118, 142, 1)",
          "secondaryColor": "#1E3D3B",
          "secondaryTextColor": "#283035",
          "tertiaryBackgroundColor": "#F7F7F7",
          "textPrimary": "#1E2428",
          "white": "#FFFFFF",
        },
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "breakpoints": Array [
          "500px",
          "52em",
          "1242px",
        ],
        "colors": Object {
          "bgPrimary": "#1E2428",
          "bgSecondary": "#1E3D3B",
          "borderTextArea": "#979797",
          "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
          "disabledBackgroundColor": "#D6D6D6",
          "disabledTextColor": "#9B9B9B",
          "divider": "#EBEBEB",
          "errorRedColor": "#C03030",
          "hoverTextColor": "#108E72",
          "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
          "paddingTop": "2.5em",
          "primary": "#0BA986",
          "primaryColor": "#0BA986",
          "primaryDark": "#11856B",
          "primaryTextColor": "#1E2428",
          "secondary": "#11856B",
          "secondaryBlue": "rgb(81, 123, 162, 1)",
          "secondaryBlueDark": "rgb(81, 118, 142, 1)",
          "secondaryColor": "#1E3D3B",
          "secondaryTextColor": "#283035",
          "tertiaryBackgroundColor": "#F7F7F7",
          "textPrimary": "#1E2428",
          "white": "#FFFFFF",
        },
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "fontSizes": Array [
          11,
          13,
          14,
          16,
          18,
          20,
          22,
          28,
          32,
        ],
        "fontWeight": Array [
          300,
          "normal",
          500,
          700,
        ],
        "hoverTextColor": "#108E72",
        "padContent": "
    max-width: 1242px;
    margin-left: auto;
    margin-right: auto;
    @media (min-width: 1242px) {
      min-width: 1242px;
    }
    @media (max-width: 1242px) {
      min-width: 90%;
    }
  ",
        "padding": "calc((100vw - 1242px) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "space": Array [
          0,
          10,
          15,
          30,
          60,
        ],
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      }
    }
  >
    <CuratedCategory
      background="https://test.com/image.png"
      category="test"
      description="test"
      linkComponent={[Function]}
    >
      <CuratedCategory__Container
        background="https://test.com/image.png"
      >
        <StyledComponent
          background="https://test.com/image.png"
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "CuratedCategory__Container-sc-7fcfjn-0",
                "isStatic": false,
                "lastClassName": "cILmbH",
                "rules": Array [
                  "display:flex;z-index:1;position:relative;flex-direction:column;width:",
                  "288",
                  "px;height:90px;background:",
                  [Function],
                  ";cursor:pointer;border-radius:4px;@media (max-width:",
                  [Function],
                  "){height:60px;width:170px;> a{z-index:9001;}> a > div > *:last-child{display:none;}}",
                ],
              },
              "displayName": "CuratedCategory__Container",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "CuratedCategory__Container-sc-7fcfjn-0",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="CuratedCategory__Container-sc-7fcfjn-0 cILmbH"
          >
            <ForwardRef
              asBackground={true}
              borderRadius="4px"
              height="90px"
              image="https://test.com/image.png"
              mobileHeight="60px"
              mobileWidth="170px"
              overlay={
                Object {
                  "color": "#1E2428",
                  "opacity": 0.7,
                }
              }
              width="288px"
            >
              <InView
                rootMargin="200px 0px"
                threshold={0}
                triggerOnce={true}
              >
                <Image__Img
                  asBackground={true}
                  borderRadius="4px"
                  height="90px"
                  image="https://test.com/image.png"
                  inView={false}
                  mobileHeight="60px"
                  mobileWidth="170px"
                  overlay={
                    Object {
                      "color": "#1E2428",
                      "opacity": 0.7,
                    }
                  }
                  width="288px"
                >
                  <StyledComponent
                    asBackground={true}
                    borderRadius="4px"
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "Image__Img-sc-1jpuda7-0",
                          "isStatic": false,
                          "lastClassName": "jlLeQT",
                          "rules": Array [
                            "height:",
                            [Function],
                            ";width:",
                            [Function],
                            ";",
                            [Function],
                            ";background:",
                            [Function],
                            ";background-size:cover;opacity:",
                            [Function],
                            ";transition:opacity ",
                            [Function],
                            "s;",
                            [Function],
                            ";",
                            [Function],
                            ";position:",
                            [Function],
                            ";",
                            [Function],
                            ";",
                            [Function],
                            ";",
                            [Function],
                            " ",
                            [Function],
                            ";",
                          ],
                        },
                        "displayName": "Image__Img",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "Image__Img-sc-1jpuda7-0",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={[Function]}
                    height="90px"
                    image="https://test.com/image.png"
                    inView={false}
                    mobileHeight="60px"
                    mobileWidth="170px"
                    overlay={
                      Object {
                        "color": "#1E2428",
                        "opacity": 0.7,
                      }
                    }
                    width="288px"
                  >
                    <div
                      className="Image__Img-sc-1jpuda7-0 jlLeQT"
                      height="90px"
                      width="288px"
                    />
                  </StyledComponent>
                </Image__Img>
              </InView>
            </ForwardRef>
            <CuratedCategory__Content>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "CuratedCategory__Content-sc-7fcfjn-1",
                      "isStatic": false,
                      "lastClassName": "kdSZBI",
                      "rules": Array [
                        "display:flex;z-index:9001;width:100%;position:absolute;flex-direction:column;text-overflow:ellipsis;padding:",
                        [Function],
                        "px;overflow-x:hidden;>:first-child{margin-bottom:",
                        [Function],
                        "px;}",
                      ],
                    },
                    "displayName": "CuratedCategory__Content",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "CuratedCategory__Content-sc-7fcfjn-1",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <div
                  className="CuratedCategory__Content-sc-7fcfjn-1 kdSZBI"
                >
                  <Typography__component
                    color="white"
                    textTransform="uppercase"
                  >
                    <StyledComponent
                      color="white"
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "Typography__component-sc-36xkf6-0",
                            "isStatic": false,
                            "lastClassName": "jMarAF",
                            "rules": Array [
                              "font-size:11px;font-weight:bold;text-transform:uppercase;color:",
                              [Function],
                              ";text-align:",
                              [Function],
                              ";",
                              [Function],
                              " ",
                              [Function],
                            ],
                          },
                          "displayName": "Typography__component",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "Typography__component-sc-36xkf6-0",
                          "target": "span",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      textTransform="uppercase"
                    >
                      <span
                        className="Typography__component-sc-36xkf6-0 jMarAF"
                        color="white"
                      >
                        test
                      </span>
                    </StyledComponent>
                  </Typography__component>
                  <Typography__component
                    color="white"
                  >
                    <StyledComponent
                      color="white"
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "Typography__component-sc-36xkf6-4",
                            "isStatic": false,
                            "lastClassName": "dvuWvG",
                            "rules": Array [
                              "font-size:14px;font-weight:normal;letter-spacing:-0.1px;margin-bottom:",
                              [Function],
                              "px;line-height:18px;color:",
                              [Function],
                              ";",
                              ";",
                              [Function],
                              ";",
                            ],
                          },
                          "displayName": "Typography__component",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "Typography__component-sc-36xkf6-4",
                          "target": "span",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <span
                        className="Typography__component-sc-36xkf6-4 dvuWvG"
                        color="white"
                      >
                        test
                      </span>
                    </StyledComponent>
                  </Typography__component>
                </div>
              </StyledComponent>
            </CuratedCategory__Content>
          </div>
        </StyledComponent>
      </CuratedCategory__Container>
    </CuratedCategory>
  </ThemeProvider>
</ThemeProvider>
`;
